client/app/bundles/course/user-invitations/operations.ts

```diff
@@ -129,3 +130,62 @@ export function toggleRegistrationCode(shouldEnable: boolean): Operation {
         );
       });
 }
+
+export const splitEntries = (input: string): string[] => {
+  const entries: string[] = [];
+  let currentEntry = '';
+  let inQuotes = false;
+  for (let i = 0; i < input.length; i++) {
+    const char = input[i];
+    if (char === '"') {
+      inQuotes = !inQuotes;
+      currentEntry += char;
+    } else if ((char === ',' || char === ';') && !inQuotes) {
+      if (currentEntry.trim()) {
+        entries.push(currentEntry.trim());
+      }
+      currentEntry = '';
+    } else {
+      currentEntry += char;
+    }
+  }
+  if (currentEntry.trim()) {
+    entries.push(currentEntry.trim());
+  }
+  return entries;
+};
+
+export const parseEmails = (
+  input: string,
+): { results: InvitationEntry[]; errors: string[] } => {
+  const regex = /^(?:"([^"]+)"|([^"<]+))?\s*<([^>]+)>$|^([^"<\s]+@[^\s,;<>]+)$/;
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const results: InvitationEntry[] = [];
+  const errors: string[] = [];
+
+  const entries = splitEntries(input);
+
+  entries.forEach((entry) => {
+    const match = regex.exec(entry);
+    if (match) {
+      if (match[3]) {
+        const name = (match[1] || match[2] || '').trim();
```

Comment:
Can we simplify the regex? Why so many cases?

Reply:
since indeed we need to handle quite many cases for email parsing:
- entry only has emails (in which name should be equal to email)
- entry has "Name" <email> (need to parse between double quotes so that it's not getting assumed as delimiter)
- entry has Name <email>

Reply:
See the updated logic.